web/client/components/map/cesium/Map.jsx

```diff
@@ -190,6 +190,19 @@ class CesiumMap extends React.Component {
             };
             this.setView(position);
         }
+
+        if (prevProps && (!isEqual(this.props.mapOptions.showSkyAtmosphere, prevProps?.mapOptions?.showSkyAtmosphere))) {
+            this.map.scene.skyAtmosphere.show = this.props.mapOptions.showSkyAtmosphere;
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.showGroundAtmosphere, prevProps?.mapOptions?.showGroundAtmosphere))) {
+            this.map.scene.globe.showGroundAtmosphere = this.props.mapOptions.showGroundAtmosphere;
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.enableFog, prevProps?.mapOptions?.enableFog))) {
+            this.map.scene.fog.enabled = this.props.mapOptions.enableFog;
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.depthTestAgainstTerrain, prevProps?.mapOptions?.depthTestAgainstTerrain))) {
+            this.map.scene.globe.depthTestAgainstTerrain = this.props.mapOptions.depthTestAgainstTerrain;
+        }
```

Comment:
Please update this branch with the latest master and verify if we need to add map.scene.requestRender(); after this scene updates to visualize the changes in the viewer

related to https://github.com/geosolutions-it/MapStore2/commit/3df6f48a01f34b4dbd9ef444a0a30b6ae819187f

Reply:
It is not needed, it seems that the scene is updated (and rendered) right after property reassignment.